<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>928</width>
    <height>732</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-image: url(:/newPrefix/Optimized-AI_image_Google-1024x576.jpg);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="frame_3">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>50</y>
      <width>581</width>
      <height>271</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="AddGoalsButton">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>160</y>
       <width>100</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Add goal(s)</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="StartNodeBox">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>40</y>
       <width>113</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>120</y>
       <width>111</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Add goal(s) node</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="GoalNodeBox">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>140</y>
       <width>113</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="AddStartNodeButton">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>60</y>
       <width>100</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Add Start</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>20</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Add start node</string>
     </property>
    </widget>
    <widget class="QPushButton" name="DrawGraphButton">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>220</y>
       <width>100</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>DrawGraph</string>
     </property>
    </widget>
    <widget class="QPushButton" name="ResetStartButton">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>60</y>
       <width>100</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>ResetStart</string>
     </property>
    </widget>
    <widget class="QPushButton" name="ResetGoalButton">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>160</y>
       <width>100</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>ResetGoal</string>
     </property>
    </widget>
    <zorder>AddGoalsButton</zorder>
    <zorder>StartNodeBox</zorder>
    <zorder>GoalNodeBox</zorder>
    <zorder>AddStartNodeButton</zorder>
    <zorder>label_6</zorder>
    <zorder>label_7</zorder>
    <zorder>DrawGraphButton</zorder>
    <zorder>ResetStartButton</zorder>
    <zorder>ResetGoalButton</zorder>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>350</y>
      <width>371</width>
      <height>191</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QComboBox" name="AlgoSelectBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>201</width>
       <height>32</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>BreadthFirstSearch</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>DepthFirstSearch</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>UniformCostSearch</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>IterativeDeepeningSearch</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>GreedySearch</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>AStarSearch</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>111</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>.AppleTraditionalChineseFont</family>
       <weight>75</weight>
       <italic>true</italic>
       <bold>true</bold>
       <underline>false</underline>
      </font>
     </property>
     <property name="text">
      <string>Search Method</string>
     </property>
    </widget>
    <widget class="QPushButton" name="ResetGraphButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>100</y>
       <width>101</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>ResetGraph</string>
     </property>
    </widget>
    <widget class="QPushButton" name="DrawPathButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>70</y>
       <width>100</width>
       <height>32</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>DrawPath</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>380</y>
      <width>321</width>
      <height>151</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="AddHeuristicButton">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>90</y>
       <width>125</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Add heuristic</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>40</y>
       <width>58</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Heuristic</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="HeuristicBox">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>60</y>
       <width>125</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="CurrentNodeHeuristicBox">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>113</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>58</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Node</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>125</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Add heuristic</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>151</width>
      <height>351</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Add edge</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>From</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="FromBox">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_10">
       <property name="text">
        <string>To</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="ToBox">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_11">
       <property name="text">
        <string>Weight</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="WeightBox"/>
     </item>
     <item>
      <widget class="QPushButton" name="AddEdgeButton">
       <property name="text">
        <string>Add edge</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Type</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="DirectionBox">
       <item>
        <property name="text">
         <string>Directed</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>UnDirected</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="SetTypeButton">
       <property name="text">
        <string>Set type</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>928</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources>
  <include location="background.qrc"/>
 </resources>
 <connections/>
</ui>
